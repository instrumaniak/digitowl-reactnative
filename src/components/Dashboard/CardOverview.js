/**
 *   Dashboard section: Overview
 */

import React, { PureComponent } from 'react'
import { View, StyleSheet } from 'react-native'

import {
  Card,
  CardItem,
  Text
} from 'native-base'

import {
  LineChart,
  XAxis,
  YAxis,
  Grid
} from 'react-native-svg-charts'

import { human } from 'react-native-typography'
import { Circle } from 'react-native-svg'
import { colors } from '../../config'

const data = [ 50, 10, 40, 95, -4, -24, 85, 91, 35, 53, -53, 24, 50, -20, -80 ]

const Decorator = ({ x, y, data }) => {
  return data.map(( value, index) => (
    <Circle
      key={index}
      cx={ x(index) }
      cy={ y(value)}
      r={ 4 }
      stroke={ colors.grey }
      strokeWidth={3}
      fill={colors.whiteSmoke}
    />
  ))
}

class CardOverview extends PureComponent {
  render() {
    return (
      <Card style={{ elevation: 2 }}>
        <CardItem>
          <Text style={styles.cardHeaderText}>Overview</Text>
        </CardItem>
        <CardItem>
          <View style={styles.btnContainer}>
            <View style={[styles.overviewButton, styles.oBtnActive]}>
              <Text style={styles.obtnHeading}>Total Wealth</Text>
              <Text style={styles.obtnValue}>+34 628 USD</Text>
            </View>
            <View style={styles.overviewButton}>
              <Text style={styles.obtnHeading}>Monthly Cash Flow</Text>
              <Text style={styles.obtnValue}>+257 USD</Text>
            </View>
          </View>
        </CardItem>

        {/* Charts */}
        <CardItem>
          <View style={{ flexDirection: 'row'}}>
            <YAxis
              data={data}
              style={{ marginBottom: 30, marginRight: 10 }}
              contentInset={{ top: 10, bottom: 10 }}
              svg={{
                ...human.caption2Object,
                fill: colors.grey1,
              }}
            />
            <View style={{ flex: 1}}>
              <LineChart
                data={data}
                style={{ height: 200 }}
                svg={{stroke: colors.grey1, strokeOpacity: 0.6, strokeWidth: 3 }}
                contentInset={{ top: 10, bottom: 10 }}
              >
                <Decorator />
                <Grid svg={{strokeOpacity: 0.5, stroke: colors.grey}}/>
              </LineChart>
              <XAxis
                style={{ marginHorizontal: -10, height: 30}}
                data={data}
                contentInset={{ left: 10, right: 10 }}
                formatLabel={(value, index) => index}
                svg={{ ...human.caption2Object, fill: colors.grey1 }}
              />
            </View>
          </View>
        </CardItem>
      </Card>
    )
  }
}

const styles = StyleSheet.create({
  cardHeaderText: {
    ...human.headlineObject,
    color: colors.grey2
  },
  btnContainer: {
    flex: 1,
    flexDirection: 'row'
  },
  btnLight: {
    backgroundColor: 'white',
    flex: 1,
    marginHorizontal: 5,
    justifyContent: 'center'
  },
  overviewButton: {
    flex: 1,
    alignItems: 'center',
    padding: 5
  },
  oBtnActive: {
    backgroundColor: 'rgba(0,0,0,0.05)',
    borderRadius: 5
  },
  obtnHeading: {
    ...human.bodyObject,
    color: colors.grey1,
    marginBottom: 5,
  },
  obtnValue: {
    ...human.title3Object,
    color: colors.grey2,
  }
})

export default CardOverview
