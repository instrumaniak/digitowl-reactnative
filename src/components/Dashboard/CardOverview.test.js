/**
 *   Tests for component: CardOverview
 */

import React from 'react'
import CardOverview from './CardOverview'
import renderer from 'react-test-renderer'

describe('CardOverview component', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<CardOverview />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
