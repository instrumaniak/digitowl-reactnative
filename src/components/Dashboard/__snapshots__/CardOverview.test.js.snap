// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardOverview component Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "elevation": 2,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#4B5358",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "600",
            "letterSpacing": -0.408,
            "lineHeight": 22,
          },
        ]
      }
      uppercase={false}
    >
      Overview
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.05)",
            "borderRadius": 5,
            "flex": 1,
            "opacity": 1,
            "padding": 5,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#5B6267",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "400",
                  "letterSpacing": -0.408,
                  "lineHeight": 22,
                  "marginBottom": 5,
                },
              ]
            }
            uppercase={false}
          >
            Total Wealth
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#4B5358",
                  "fontFamily": "System",
                  "fontSize": 20,
                  "fontWeight": "400",
                  "letterSpacing": 0.361328,
                  "lineHeight": 25,
                },
              ]
            }
            uppercase={false}
          >
            +34 628 USD
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
            "padding": 5,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#5B6267",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "400",
                  "letterSpacing": -0.408,
                  "lineHeight": 22,
                  "marginBottom": 5,
                },
              ]
            }
            uppercase={false}
          >
            Monthly Cash Flow
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#4B5358",
                  "fontFamily": "System",
                  "fontSize": 20,
                  "fontWeight": "400",
                  "letterSpacing": 0.361328,
                  "lineHeight": 25,
                },
              ]
            }
            uppercase={false}
          >
            +257 USD
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginBottom": 30,
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "transparent",
                "fontSize": 11,
              }
            }
          >
            -20
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "height": 250,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "height": 30,
              "marginHorizontal": -10,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "transparent",
                  "fontSize": 11,
                }
              }
            >
              0
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
